### ISSUE TITLE
Native method, array update

### ISSUE THREAD

**Issue by derRinat** [balderdashy/sails-mongo#202]
Hi guys, 
so i try to update values of document field, which is array, with "native" method 
```javascript
this.native(function(err, collection) {
    collection.update(
        {id: game, 'players.index': data.seat}, 
        {'$set': {
                        'players.$.player_name': data.user,
                        'players.$.player_id': data.user_id
                 }
        },
        function(err, docs) {
            if(err) { return callback(false);}
            return callback(docs);
         }
    );
});
```
unfortunately, it doesn't work for me.
Mongo query will be executed as it expected and it updates document:

```javascript
db.collection.update({ _id: new ObjectId('545e2ddb873765459cd46d4e'), 'players.index': 2 }, { '$set': { 'players.$.player_name': 'mike', 'players.$.player_id': '54516af98faf7e2afa665f19' }})
```

Could somebody suggest me good solution? Thanks!

**Comment by sailsbot** [comment]
Thanks for posting, @derRinat.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
